auth.go

```diff
@@ -436,12 +461,23 @@ func (x *Central) GetTokenFromIdentityPassword(identity, password string) (*Toke
 // Create a new session. Returns a session key, which can be used in future to retrieve the token.
 // The internal session expiry is controlled with the member NewSessionExpiresAfter.
 // The session key is typically sent to the client as a cookie.
-func (x *Central) Login(username, password string) (sessionkey string, token *Token, err error) {
+func (x *Central) Login(username, password string, address string) (sessionkey string, token *Token, err error) {
 	user, authErr := x.authenticate(username, password)
 	if authErr != nil {
+
 		err = authErr
-		x.Stats.IncrementInvalidPasswords(x.Log)
 
+		if err == ErrInvalidPassword {
```

Comment:
This check cannot be performed here, because the user doesn't need to try to Login in order to check if the password is correct. He can also try to performance a once-off authorization. So all of this account locking stuff needs to happen inside authorize().